.github/workflows/clean_up_pr.yaml

```diff
@@ -0,0 +1,66 @@
+name: "Pull Request Clean Up"
+
+on:
+  pull_request:
+    branches: main
+    types: [unlabeled, closed]
+
+permissions:
+  id-token: write
+  contents: read
+  deployments: write
+
+jobs:
+  clean-up:
+    if: |
+      (github.event.action == 'unlabeled' && github.event.label.name == 'deploy') ||
+      (github.event.action == 'closed' && contains(github.event.pull_request.labels.*.name, 'deploy'))
```

Comment:
Have you tested the on close event yet? Maybe we can with #347?

Reply:
Yup, already tested. We can also test here too.